src/pydisagg/ihme/helper.py

```diff
@@ -0,0 +1,161 @@
+from .age_var import rename_dict_dis
+import matplotlib.pyplot as plt
+import numpy as np
+
+plt.style.use("ggplot")
+
+
+def rename_df(frozen_df, rename_dict=rename_dict_dis, drop=True):
+    """
+    Renames columns of a DataFrame based on a given dictionary and optionally drops columns.
+
+    Parameters:
+        frozen_df (DataFrame): The input DataFrame to be renamed.
+        rename_dict (dict): A dictionary mapping old column names to new column names.
+        drop (bool, optional): Whether to drop columns not present in the rename_dict.
+                               Defaults to True.
+
+    Returns:
+        return_df: The renamed dataframe
+        frozen_df: The original dataframe with 'row_id' column added
+    """
+    # Create a copy of the input DataFrame to avoid changing it in place
+    df = frozen_df.copy()
+    frozen_df_copy = frozen_df.copy()
+
+    df.insert(0, "row_id", range(1, len(df) + 1))
+    frozen_df_copy.insert(0, "row_id", range(1, len(frozen_df_copy) + 1))
+
+    return_df = df.rename(columns=rename_dict)
+    if drop:
+        return_df = return_df[list(rename_dict.values())]
+    return_df.insert(0, "row_id", range(1, len(return_df) + 1))
+
+    return return_df, frozen_df_copy
```

Comment:
I don't think you need to copy `frozen_df` twice. Since you never use `inplace=True`, `df` will be exactly the same with `frozen_df_copy` at the end. May I ask why do we need `row_id`?

Reply:
Row_id is used to map all the columns back to the original dataset. Often users have submitted datasets with col names that overlap with ones we need later so i drop them at the start. You're right, the copy is extra now reminensce of troubleshooting :/